### PR TITLE
[NFC] Miscellaneous Cleanups To Clang Module Loading

### PR DESCRIPTION
**Pull request by CodaFi**
* Drop the overlay-forcing parameter. That's already happening when we populate the import cache.
* Fold code duplicating `getWrapperForModule` in clang module loading.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
@swift-ci test source compatibility

**Comment by CodaFi** [comment]
@swift-ci test source compatibility release

**Comment by CodaFi** [comment]
@swift-ci clean test source compatibility release

**Comment by CodaFi** [comment]
Ah, the joys of non-deterministic diagnostics

@swift-ci test macOS platform

**Comment by CodaFi** [comment]
@swift-ci clean test macOS platform

**Comment by CodaFi** [comment]
Fech, everything fell over.

@swift-ci clean smoke test

**Comment by CodaFi** [comment]
@swift-ci test source compatibility

File: lib/ClangImporter/ClangImporter.cpp [hunk 1]
```diff
@@ -1800,59 +1799,30 @@ ModuleDecl *ClangImporter::Implementation::loadModule(
...
-    wrapperUnit =
-      new (SwiftContext) ClangModuleUnit(*result, *this, clangModule);
-    result->addFile(*wrapperUnit);
-    SwiftContext.getClangModuleLoader()
-        ->findOverlayFiles(importLoc, result, wrapperUnit);
-    cacheEntry.setPointerAndInt(wrapperUnit, true);
-
-    // Force load overlays for all imported modules.
-    // FIXME: This forces the creation of wrapper modules for all imports as
-    // well, and may do unnecessary work.
```
Comment (davidungar) [inline lib/ClangImporter/ClangImporter.cpp:1840]: Nice to rid of all this stuff. I don't know the ClangImporter well enough to understand how we can afford to get rid of it, but I'll take your word for it.

**Review by davidungar — APPROVED** [review]
Overall, looks nice and clean. I don't know enough to vouch for its correctness, though.

**Comment by CodaFi** [comment]
⛵

**Pull request merged**